source/services/encryption/EncryptionService.ts

```diff
@@ -53,3 +63,53 @@ export async function decryptWithAesKey(user: User, cipher: string): Promise<str
 
   return Aes.decrypt(cipher, aesEncryptor.aesKey, aesEncryptor.initializationVector);
 }
+
+export async function decryptFormAnswers(user: UserInterface, forms: FormsInterface) {
+  if (forms.encryption.type === 'privateAesKey') {
+    const { encryptedAnswers } = forms.answers;
+    const decryptedAnswers = await decryptWithAesKey(user, encryptedAnswers);
+
+    forms.answers = JSON.parse(decryptedAnswers);
+    forms.encryption.type = 'decrypted';
+
+    return forms;
+  }
+}
+
+export async function setupSymmetricKey(user, forms) {
+  const otherUserPersonalNumber = Object.keys(forms.encryption.publicKey.publicKeys).filter(
+    (key) => key !== user.personalNumber
+  )[0];
+
+  const otherUserPublicKey = getPublicKeyInForm(otherUserPersonalNumber, forms);
+  let ownPublicKey = getPublicKeyInForm(user.personalNumber, forms);
+
+  if (!ownPublicKey) {
+    const privateKey = await createAndStorePrivateKey(user, forms);
+    ownPublicKey = await getPseudoKey(
+      forms.encryption.publicKey.G,
+      privateKey,
+      forms.encryption.publicKey.P
+    );
+
+    forms.encryption.publicKey.publicKeys[user.personalNumber] = ownPublicKey;
+  }
+
+  if (typeof ownPublicKey !== 'undefined' && typeof otherUserPublicKey !== 'undefined') {
+    // Generate symmetric key.
```

Comment:
This comment implies that we could extract it to a separeate function, named inspired by the comment :)